base/string.jl

```diff
@@ -1251,6 +1251,79 @@ print_shell_escaped(io::IO) = nothing
 
 shell_escape(args::AbstractString...) = sprint(print_shell_escaped, args...)
 
+# Functions for Windows shell escaping (mode can be :cmd or :powershell)
+function print_cmdshell_word(io::IO, word::AbstractString, mode=:cmd)
```

Comment:
i'm inclined to suggest adding making `mode` a keyword arg option (with a default of `:sh`) and merging all three of the `print_..._word` and `print_..._escaped` functions. anyone else want to weigh in here?